### PR TITLE
Resident Status Style

### PR DESCRIPTION
**Pull request by owen-sellner**
## Notion task link

N/A

## Implementation description
* Added ResidentStatus enum to `ResidentTypes.ts` to ensure status string values are consistant 
* Updated resident status function in `ResidentDirectoryTable.tsx` to include "Future" as a status
* Added getStatusColor method to `ResidentDirectoryTable.tsx` to convert the status to color 

## Steps to test
1. Create three new residents (use `create-residents.sh`)
2. Edit the Start Date and End Date for the residents and verify that the status is correct

## What should reviewers focus on?
* Are the colours for the statuses okay? (They match the employee directory)

## Screenshot
image

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [x] If I have made API changes, I have updated the [REST API Docs](https://www.notion.so/uwblueprintexecs/REST-Endpoints-05ce60312bb943439dfda42bb1318536)
- [x] IF I have made changes to the db/models, I have updated the [Data Models Page](https://www.notion.so/uwblueprintexecs/Data-Models-760f8aa06b244eb0842c079ad77987b0)
- [x] I have documented this PR in the [Production Release Page](https://www.notion.so/uwblueprintexecs/fe28a97bfa5648d3bc3795b32b694acd?v=5565cfb35dcc45bb84f6528cb09517cd)
- [x] I have updated other Docs as needed

### CODE REVIEW

File: frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx [hunk 1]
```diff
@@ -67,7 +91,7 @@ const ResidentDirectoryTable = ({
   getRecords,
   countResidents,
 }: Props): React.ReactElement => {
-  const { authenticatedUser, setAuthenticatedUser } = useContext(AuthContext);
```
Comment (connor-bechthold) [inline frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx:70]: was this unused

**Review by connor-bechthold — CHANGES_REQUESTED** [review]
With these bubbles I think it looks better if it's centered in the column similar to how it's done in the Employee Directory
Screenshot 2023-12-02 at 9 26 22 PM

Also, there's a bug if you set the end date to the same date as the current date - it shows `Past` instead of `Current`. I believe this has to do with the fact that the `new Date()` you're making has hours associated with it; if you set the hours, minutes, and seconds, to 0 it should fix this. 

Colours on the status looks great tho, frontend god strikes again 💯

File: frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx (hunk 1 above)
Comment (owen-sellner) [reply frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx:70]: setAuthenicatedUser is not used but authenticatedUser is used

File: frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx [hunk 2]
```diff
@@ -35,19 +35,44 @@ type Props = {
...
+  const currentDate = new Date();
+  currentDate.setHours(0,0,0,0);
   if (resident.dateLeft != null) {
     const endDateObj = convertToDate(resident.dateLeft);
     endDate = getFormattedDateAndTime(endDateObj, true);
+    if (endDateObj < currentDate) {
+      status = ResidentStatus.PAST;
+    }
+  }
+  if (currentDate < startDateObj) {
```
Comment (phamkelly17) [inline frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx:73]: nit: could make this an else if lol

Comment (phamkelly17) [inline frontend/src/components/pages/ResidentDirectory/ResidentDirectoryTable.tsx:73]: nit: could make this an else if lol

**Review by phamkelly17 — APPROVED** [review]
lgtm alpha

**Review by connor-bechthold — APPROVED** [review]
LGTM

**Pull request merged**